FHEM/00_SIGNALduino.pm

```diff
@@ -318,12 +317,83 @@ my %sets;
   							],
   "cc1101_patable_433" 	=>	['-10_dBm,-5_dBm,0_dBm,5_dBm,7_dBm,10_dBm', \&cc1101::SetPatable ],
   "cc1101_patable_868" 	=> 	['-10_dBm,-5_dBm,0_dBm,5_dBm,7_dBm,10_dBm', \&cc1101::SetPatable ],
+  "cc1101_reg"					=> 	[ 'textFieldNL', sub {
+																my ($hash, @a) = @_;
+																my $arg = $a[1];
+																my @args = split(' ', $arg);
+																my $argadr;
+																
+																foreach my $argcmd (@args) {
+																	if ($argcmd =~ /^[0-9A-Fa-f]{4}$/) {
+																		## check allowed register position
+																		return "ERROR: unknown register position ".substr($argcmd,0,2) if ! ($cc1101_register{substr($argcmd,0,2)});
+																		
+																		$argadr = hex(substr($argcmd,0,2)) + 2;
+																		$argcmd = sprintf("W%02X%s",$argadr,substr($argcmd,2,2));
+																		$hash->{logMethod}->($hash->{NAME}, 4, "$hash->{NAME}: Set, cc1101_reg $argcmd");
+																		SIGNALduino_AddSendQueue($hash,$argcmd);
+																	} else {
+																		return "ERROR: wrong register value $argcmd, only hexadecimal ​​four digits allowed"
+																	}
+																}
+																SIGNALduino_WriteInit($hash);
+															}
+														],
 );
 
 
-
 my @modformat = ("2-FSK","GFSK","-","ASK/OOK","4-FSK","-","-","MSK"); # modulation format
 
+my %cc1101_register = (		# for get ccreg 99 and set cc1101_reg
```

Comment:
Die Deklaration der Variable muss weiter nach oben, damit es in der sub weiter oben auch bekannt ist :)

Reply:
hab es gelöst